diff --git a/drivers/net/ethernet/broadcom/bnx2x/Makefile b/drivers/net/ethernet/broadcom/bnx2x/Makefile
index 2523cfc7527d..1fb5124443b7 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/Makefile
+++ b/drivers/net/ethernet/broadcom/bnx2x/Makefile
@@ -5,5 +5,8 @@
 
 obj-$(CONFIG_BNX2X) += bnx2x.o
 
+ifeq ($(ARCH),ia64)
+        CFLAGS_bnx2x_sp.o += -fno-var-tracking
+endif
 bnx2x-y := bnx2x_main.o bnx2x_link.o bnx2x_cmn.o bnx2x_ethtool.o bnx2x_stats.o bnx2x_dcb.o bnx2x_sp.o bnx2x_self_test.o
 bnx2x-$(CONFIG_BNX2X_SRIOV) += bnx2x_vfpf.o bnx2x_sriov.o
diff --git a/net/ipv4/Makefile b/net/ipv4/Makefile
index ec36d2ec059e..17f431b6a205 100644
--- a/net/ipv4/Makefile
+++ b/net/ipv4/Makefile
@@ -3,6 +3,9 @@
 # Makefile for the Linux TCP/IP (INET) layer.
 #
 
+ifeq ($(ARCH),ia64)
+        CFLAGS_fib_semantics.o += -fno-var-tracking
+endif
 obj-y     := route.o inetpeer.o protocol.o \
 	     ip_input.o ip_fragment.o ip_forward.o ip_options.o \
 	     ip_output.o ip_sockglue.o inet_hashtables.o \
diff --git a/net/ipv6/Makefile b/net/ipv6/Makefile
index d283c59df4c1..160f6b645f75 100644
--- a/net/ipv6/Makefile
+++ b/net/ipv6/Makefile
@@ -5,6 +5,10 @@
 
 obj-$(CONFIG_IPV6) += ipv6.o
 
+ifeq ($(ARCH),ia64)
+	CFLAGS_raw.o += -fno-var-tracking
+endif
+
 ipv6-y :=	af_inet6.o anycast.o ip6_output.o ip6_input.o addrconf.o \
 		addrlabel.o \
 		route.o ip6_fib.o ipv6_sockglue.o ndisc.o udp.o udplite.o \
