diff --git a/fs/nfs/Makefile b/fs/nfs/Makefile
index 22d11fdc6deb..9d6f9b9ef2a5 100644
--- a/fs/nfs/Makefile
+++ b/fs/nfs/Makefile
@@ -6,6 +6,9 @@
 obj-$(CONFIG_NFS_FS) += nfs.o
 
 CFLAGS_nfstrace.o += -I$(src)
+ifeq ($(ARCH),ia64)
+	CFLAGS_inode.o += -fno-var-tracking
+endif
 nfs-y 			:= client.o dir.o file.o getroot.o inode.o super.o \
 			   io.o direct.o pagelist.o read.o symlink.o unlink.o \
 			   write.o namespace.o mount_clnt.o nfstrace.o \
