From 39442fd58da7549c7fee7ae33046c19b7c7a7a96 Mon Sep 17 00:00:00 2001
From: Johnny Mnemonic <jm@machine-hall.org>
Date: Sat, 4 Jan 2025 11:33:17 +0100
Subject: [PATCH 48/96] ia64: Extend processor feature flags according to SDM
 2.3

---
 arch/ia64/kernel/setup.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/ia64/kernel/setup.c b/arch/ia64/kernel/setup.c
index 52e0334c5779..d0369d2af51b 100644
--- a/arch/ia64/kernel/setup.c
+++ b/arch/ia64/kernel/setup.c
@@ -661,9 +661,12 @@ show_cpuinfo (struct seq_file *m, void *v)
 		unsigned long mask;
 		const char *feature_name;
 	} feature_bits[] = {
-		{ 1UL << 0, "branchlong" },
-		{ 1UL << 1, "spontaneous deferral"},
-		{ 1UL << 2, "16-byte atomic ops" }
+		{ 1UL <<  0, "branchlong" },
+		{ 1UL <<  1, "spontaneous deferral" },
+		{ 1UL <<  2, "16-byte atomic ops" },
+		{ 1UL <<  3, "resource utilization counter" },
+		{ 1UL << 32, "clz" },
+		{ 1UL << 33, "mpy4/mphysl4" }
 	};
 	char features[128], *cp, *sep;
 	struct cpuinfo_ia64 *c = v;
-- 
2.25.1

